src/Game/Systems/PlayerDeathSystem.cpp

```diff
@@ -33,10 +35,10 @@ void PlayerDeathSystem::createDeathEffect(EntityWrapper player)
 
     //components that we need from player
     auto playerModel = player.FirstChildByName("PlayerModel");
-    if (!playerModel.Valid()) {
-        return;
-    }
-    if (!playerModel.HasComponent("Model") || !playerModel.HasComponent("Animation")) {
+    if (!playerModel.Valid() || !playerModel.HasComponent("Model") || !playerModel.HasComponent("Animation")) {
```

Comment:
I dont think you need !playerModel.Valid() since if it has model/animation, it will be valid.